modules/gapi/src/api/render_ocv.cpp

```diff
@@ -79,10 +79,16 @@ void poly(cv::Mat mat, std::vector<cv::Point>, cv::Scalar color, int lt, int shi
 
 void mosaic(cv::Mat mat, const cv::Rect &rect, int cellSz)
 {
-    auto mos = mat(rect);
-    cv::Mat tmp;
-    cv::resize(mos, tmp, {mos.size().width/cellSz, mos.size().height/cellSz}, 0, 0, cv::INTER_LINEAR);
-    cv::resize(tmp, mos, {rect.width, rect.height}, 0, 0, cv::INTER_LINEAR);
+    cv::Mat msc_roi = mat(rect);
+    int crop_x = msc_roi.cols - msc_roi.cols % cellSz;
```

Comment:
Maybe it's better to not handle cases where `msc_roi.cols % cellSz` != 0?

Reply:
OSD handle it

Reply:
Nevermind